### PR TITLE
(#26) fixed gearman log output for errors

### PR DESCRIPTION
**Pull request by cliffle**

### CODE REVIEW

**Review by svettwer — CHANGES_REQUESTED** [review]
Looks like a minimal change. :+1: 
Nevertheless, a test case would be great. Without a developer test (setting up sakuli, etc) I'm unable to see whether the change fixes the issue.

File: packages/sakuli-forwarder-gearman/src/gearman/submit-job.function.spec.ts [hunk 1]
```diff
@@ -83,6 +84,7 @@ describe("submit gearman job", () => {
 
     // THEN
     await expect(submittedJob).rejects.toBeUndefined();
+    expect(logger.debug).toHaveBeenCalledWith(`Received Gearman event ${event} with following data:`, argsObject);
```
Comment (svettwer) [inline packages/sakuli-forwarder-gearman/src/gearman/submit-job.function.spec.ts]:
```suggestion
    expect(logger.error).toHaveBeenCalledWith(`Received Gearman event ${event} with following data:`, argsObject);
```

File: packages/sakuli-forwarder-gearman/src/gearman/submit-job.function.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ export async function submitJob(data: GearmanData, logger: Maybe<SimpleLogger>):
             "error"
         ].forEach(evt => data.connection.on(evt, (args: any) => {
             ifPresent(logger, (log) => {
-                log.debug(`Received Gearman event: ${evt} - ${args ? args : ""}`);
+                log.debug(`Received Gearman event ${evt} with following data:`, args);
```
Comment (svettwer) [inline packages/sakuli-forwarder-gearman/src/gearman/submit-job.function.ts]:
```suggestion
                log.error(`Received Gearman event ${evt} with following data:`, args);
```

**Review by svettwer — CHANGES_REQUESTED** [review]
Thx for adding the test.

While reviewing the change from E2E, I recognized that the errors are printed to debug level, even if all of them are errors causing a promise rejection. Therefore I suggested to change the log level. It's a bit off topic but changing it now would be more efficient than raising another ticket. Or do you see any reason why we should keep it on debug?

Sample log
```
[Thu Dec 03 2020 12:23:42 GMT+0100 (GMT+01:00)] DEBUG: Received Gearman event error with following data:
Error: connect ECONNREFUSED 127.0.0.1:4730
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1145:16) {
  [stack]: 'Error: connect ECONNREFUSED 127.0.0.1:4730\n' +
    '    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1145:16)',
  [message]: 'connect ECONNREFUSED 127.0.0.1:4730',
  errno: 'ECONNREFUSED',
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 4730
}
[Thu Dec 03 2020 12:23:42 GMT+0100 (GMT+01:00)] ERROR: Failed to forward, error occured.
```

**Comment by cliffle** [comment]
Totally agree with you that it should be logged on error level. I didn't even notice that :+1:

**Review by svettwer — DISMISSED** [review]

**Review by svettwer — APPROVED** [review]

**Pull request merged**